contracts/Vault1.sol

```diff
@@ -1,6 +1,32 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+Develop a vault where users can deposit and withdraw any ERC20 token.
+*/
 contract Vault1 {
-    
+
+    event Deposited(uint256 amount);
+    event Withdrawn(uint256 amount);
+
+    address contractAddr;
+
+    constructor() {
+        contractAddr = address(this);
+    }
+
+    function deposit(address _tokenAddress, uint256 _amount) public {
+        IERC20 targetERC20 = IERC20(_tokenAddress);
+        targetERC20.transferFrom(msg.sender, contractAddr, _amount);
```

Comment:
You should keep a record of the number of tokens the user has transferred, In case someone only deposits 100 tokens but makes a request for 1000